src/components/button/button.svelte

```diff
@@ -321,43 +336,85 @@
     --icon-color: var(--color);
   }
   .leoButton.isPlain {
-    --radius: 8px;
-    --padding-x: 2px;
     --primary-color: var(--leo-button-color, var(--leo-color-primitive-primary-60));
     --color: var(--mixed-primary-color);
     --box-shadow-hover: none;
+    --bg-hover-mix: 5%;
 
     @theme (dark) {
       --primary-color: var(--leo-button-color, var(--leo-color-primitive-primary-40));
+      --bg-hover-mix: 10%;
+    }
+
+    @supports (color: color-mix(in srgb, transparent, transparent)) {
+      --bg-hover: color-mix(
+        in srgb,
+        var(--primary-color) var(--bg-hover-mix),
+        var(--background)
+      );
     }
 
     &:disabled:not(.isLoading) {
       --color: var(--leo-color-text-primary);
     }
 
     &.fab {
+      --radius: var(--leo-radius-m);
+      --bg-hover: transparent;
       --padding-y: 0;
       --padding-x: 0;
     }
   }
   .leoButton.isPlainFaint {
-    --radius: 8px;
-    --padding-x: 2px;
     --foreground: transparent;
     --primary-color: currentColor;
     --color: var(--mixed-primary-color);
     --box-shadow-hover: none;
     --icon-color: var(--mixed-primary-color);
+    --bg-hover-mix: 5%;
+
+    @theme (dark) {
+      --bg-hover-mix: 10%;
+    }
 
     @supports (color: color-mix(in srgb, transparent, transparent)) {
       --icon-color: color-mix(
         in srgb,
         var(--primary-color),
         var(--background) 30%
       );
+
+      position: relative;
+      z-index: 0;
+
+      &::before {
+        content: '';
+        width: 100%;
+        height: 100%;
+        position: absolute;
+        top: 0;
+        right: 0;
+        bottom: 0;
+        left: 0;
+        z-index: -1;
+        border-radius: var(--radius);
+        background-color: color-mix(
+          in srgb,
+          var(--primary-color) var(--bg-hover-mix),
+          var(--foreground)
+        );
+        transition: var(--default-transition);
+        opacity: 0.01;
+      }
+
+      &:hover::before {
+        opacity: 1;
+      }
```

Comment:
Note that since `plain-faint` relies on `currentColor`, transitions are working, so I'm using a pseudo element instead in order to transition the opacity of the element instead of the background color.